src/react-native/05-intro-to-testing/intro-to-testing.md

```diff
@@ -2,23 +2,84 @@
 
-TODO
+How do we know when our code is working correctly? How do we know it’s still working correctly in the future after we make changes to it?
 
-### Concept TODO
+Testing helps by verifying that given certain inputs our code generates expected outputs. So far we’ve copied existing tests to prove that we’ve completed the exercise correctly, now let’s dive in and learn about how React Native testing is done.
 
-TODO
+### Introducing React Native Testing Library
+
+[React Native Testing Library (RNTL)](https://callstack.github.io/react-native-testing-library/docs/getting-started) is a set of tools that helps us write robust tests for our React Native applications. Unlike some other testing libraries that focus on the internal state and implementation details of components, RNTL emphasizes testing the behavior of components as experienced by the end-users.
+In order to test components it provides light utility functions using React Test Renderer; most testing libraries use the DOM, but since React Native doesn't use the DOM, the React Test Renderer is necessary.
+
+Akin to the original React Testing Library, RNTL's north star is:
+
+> The more your tests resemble the way your software is used, the more confidence they can give you.
+
+RNTL encourages tests that focus on how the user interacts with the application, rather than the implementation details of the components.
+
+### Rendering and verifying a component in a test
+
+OK, you’re convinced that RNTL is a good idea, how do you use it to create tests for your React components?
+
+Let’s take a look at an example: say we want to test a component we created named `EmailInputField` that renders a form field. We expect the component will generate XML like this:
+
+```html
+<div class="form-field">
+  <label htmlFor="inputId">Email:</label>
+  <input id="inputId" type="email" value="test@example.com" />
+</div>
+```
+
+We want to test `EmailInputField` to make sure it generates the XML we expect. If you’re already familiar with testing frontend JavaScript code, the following pattern will probably be recognizable: each test consists of arguments to the `it` function provided by Jest. The first argument is a short description of what the test is verifying. The convention is that the description string takes "it" as a prefix and proceeds from there, e.g. "[it] renders the correct label and value."
+
+The second argument to `it` is a callback function that is called to run the test. Inside the callback, invoke the React Native Testing Library function `render` and pass it a single argument, the JSX for your component including props. After `render` completes, use `screen` to query the React Test Renderer and make assertions about the result.
+
+```tsx
+it("renders the correct label and value", () => {
+  render(<EmailInputField label="Email" value="test@example.com" />)
+  const label = screen.getByText("Email:", {exact: false})
+  expect(label).toBeOnTheScreen()
+})
```

Comment:
Instead of this, consider using the tests from JSX (or earlier pages) as an example.

Reply:
There's additional information that I'll be implementing from the slides it seems, so if I use the from intro to jsx as an example, I won't be able to explain userEvents and other concepts appropriately.

To be honest I'm a little confused. and may need to meet with both of you in the afternoon tomorrow. 

I was under the impression throughout the rest of the training we'd be drip feeding more information about testing information, but if I follow through with all of the React DOM testing, or those slides, the section will talk about testing features that they have not implemented yet. This could be fine but I may need a bit more clarity.